gpiozero/rotary_encoder.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+from gpiozero import Device, DigitalInputDevice, Button, CompositeDevice
+
+
+class RotaryEncoder(Device):
+    """
+    Decode mechanical rotary encoder pulses.
+
+    Connect the extrems pins of the rotary encoder (left and right pins) to any GPIO
+    and the middle pin to a ground pin. Alternatively, connect the middle pin to the
+    3V3 pin, then set *pull_up* to ``False`` in the :class:`RotaryEncoder` constructor.
+
+    The following example will print a Rotary Encoder change direction::
+
+        from gpiozero import RotaryEncoder
+
+        def change(value):
+            if value > 0:
+                print("clockwise")
+
+            else:
+                print("counterclockwise")
+
+        rotary = RotaryEncoder(13, 19)
+        rotary.when_rotated = change
+
+    Based in Pigpio implementation `Pigpio implementation <http://abyz.co.uk/rpi/pigpio/examples.html#Python_rotary_encoder_py>`_
+    and `Paul Stoffregen implementation <https://github.com/PaulStoffregen/Encoder>`_
+
+    :param int pin_a:
+        An extreme GPIO pin which the RotaryEncoder is attached to. See :ref:`pin_numbering`
+        for valid pin numbers.
+
+    :param int pin_b:
+        The another extreme GPIO pin which the RotaryEncoder is attached to. See :ref:`pin_numbering`
+        for valid pin numbers.
+
+    :param bool pull_up:
+        If ``True`` (the default), the GPIO pins will be pulled high by default.
+        In this case, connect the middle GPIO pin to ground. If ``False``, 
+        the GPIO pins will be pulled low by default. In this case,
+        connect the middle pin of the RotaryEncoder to 3V3.
+    """
+
+    when_rotated = lambda *args: None
```

Comment:
This is a class property, isn't it? Should it not be self.when_rotated defined within the __init__ method?